databricks_cli/execution_context/api.py

```diff
@@ -0,0 +1,75 @@
+# WARNING THIS FILE IS AUTOGENERATED
+#
+# Databricks CLI
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Dict, Any
+
+from databricks_cli.sdk.service import ExecutionContextService
+
+
+class ExecutionContext(object):
+    def __init__(self, api_client, cluster_id):
+        self.client = ExecutionContextService(api_client)
+        self.cluster_id = cluster_id
+        self.id = None
+
+    def __enter__(self):
+        if self.id is None:
+            self.id = self.client.create_context(cluster_id=self.cluster_id)["id"]
+
+        return self
+
+    def __exit__(self, _type, _value, _traceback):
+        self.client.delete_context(cluster_id=self.cluster_id, context_id=self.id)
+        self.id = None
+
+
+class ExecutionContextApi(object):
```

Comment:
Could you reuse the basic structure for command/context APIs from Tunnel CLI, so that it won't break, once merged? :) 
https://github.com/databricks/databricks-cli/commit/6c429d4f581fc99f54cbf9239879434310037cc7

it's removed, so probably it has to be re-added
https://github.com/databricks/databricks-cli/commit/0296117f531e430cd2f819652ec0837dce14ae98